artifactory/commands/utils/npmcmdutils.go

```diff
@@ -255,3 +257,46 @@ func PrintMissingDependencies(missingDependencies []buildinfo.Dependency) {
 		"Make sure the dependencies are available in Artifactory for this build.\n" +
 		"Deleting the local cache will force populating Artifactory with these dependencies.")
 }
+
+func BackupFile(filePath, backupPath string) (restore func() error, err error) {
+	fileInfo, err := os.Stat(filePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return createRestoreFileFunc(filePath, backupPath), nil
+		}
+		return nil, errorutils.CheckError(err)
+	}
+
+	fileMode := fileInfo.Mode()
+	if err = ioutils.CopyFile(filePath, backupPath, fileMode); err != nil {
+		return nil, err
+	}
+	log.Debug("The file", filePath, "was backed up successfully to", backupPath)
+	return createRestoreFileFunc(filePath, backupPath), nil
+}
+
+func createRestoreFileFunc(filePath, backupPath string) func() error {
+	return func() error {
+		if _, err := os.Stat(backupPath); err != nil {
+			if os.IsNotExist(err) {
+				err = os.Remove(filePath)
+				return errorutils.CheckError(err)
+			}
+			return errorutils.CheckError(errors.New(createRestoreErrorPrefix(filePath, backupPath) + err.Error()))
+		}
+
+		if err := os.Rename(backupPath, filePath); err != nil {
```

Comment:
We're not allowed to use ```s.Rename``` because it fails inside a docker container. We should use ```fileutils.MoveFile``` instead.